design/multiple-storage-types-support.md

```diff
@@ -122,112 +122,136 @@ Each storage backend can make this decision independently.
-1. If a field is added to an older version of the Group, then the field must also be added to the latest version of the Group.
+Note that it is **not a goal** to define a common abstraction that applies to the top level storage backends themselves, for instance a single `Cluster` type that covers both Ceph and Minio.
+We should not be trying to force each backend to look the same to storage admins, but instead we should focus on providing the common abstractions and implementations that storage providers can build on top of.
+This idea will become more clear in the following sections of this document.
 
 #### Proposed Custom Resources
 
-With the intent for Rook's resources to fulfill the desirable properties mentioned above and also meet the Kubernetes versioning guidelines, we propose the following API groups:
+With the intent for Rook's resources to fulfill the desirable properties mentioned above, we propose the following API groups:
 
-* `rook.io`: common abstractions that have use across multiple storage backends and types
-* `cluster.rook.io`: common types for clustered/distributed storage, such as `StorageCluster`, `Pool`, `Filesystem`, etc.
-  * `ceph.cluster.rook.io`: Ceph specific `*Spec` types that can be plugged into the common Cluster types.  This Ceph specific API group allows Ceph types to be versioned independently.
-  * `minio.cluster.rook.io`: Similar, but for Minio.
-  * `nexenta.cluster.rook.io`: Similar, but for Nexenta.
+* `rook.io`: common abstractions and implementations, in the form of `*Spec` types, that have use across multiple storage backends and types.  For example, storage scope, network information, placement, and resource usage.
+* `ceph.rook.io`: Ceph specific `Cluster` CRD type that the user can instantiate to have the Ceph controller deploy a Ceph cluster for them. This Ceph specific API group allows Ceph types to be versioned independently.
+* `minio.rook.io`: Similar, but for Minio.
+* `nexenta.rook.io`: Similar, but for Nexenta.
 
-This approach to resource definitions and versioning allows Rook to create an experience in `yaml` that would look like the following, where `StorageCluster` is the common/consistent top level resource the user creates, and backend specific configuration can be added as subsections.
+With this approach, the user experience to create a cluster would look like the following in `yaml`, where they are declaring and configuring a Ceph specific CRD type (from the `ceph.rook.io` API group), but with many common `*Spec` types that provide configuration and logic that is reusable across storage providers.
 
-`cluster.yaml`:
+`ceph-cluster.yaml`:
 
 ```yaml
-apiVersion: cluster.rook.io/v1
-kind: StorageCluster
+apiVersion: ceph.rook.io/v1
+kind: Cluster
 spec:
-  version: "myrepo/myimage:v0.1.0"
-  hostNetwork: true
+  version: "rook/ceph:v0.1.0"
```

Comment:
a few months ago we removed the container version from the cluster crd because it was causing versioning issues for users running a different version of the operator. The versioning story has been much cleaner with the operator being the same version as the cluster containers.